com.salesforce.b2eclipse.tests/src/main/java/com/salesforce/b2eclipse/importer/BuildWithSubPackageBazelImportTest.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2020, Salesforce.com, Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+ * following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+ * disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ * following disclaimer in the documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Salesforce.com nor the names of its contributors may be used to endorse or promote products
+ * derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package com.salesforce.b2eclipse.importer;
+
+import static org.junit.Assert.assertFalse;
+
+import java.util.Arrays;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test class for importing a Bazel test project that contains a sub module.
+ */
+public class BuildWithSubPackageBazelImportTest extends BaseBazelImproterTest {
+
+    public BuildWithSubPackageBazelImportTest() {
+        super.setWorkspaceRootPackage(getScanner().getProjects("projects/build-with-subpackage"));
+    }
+
+    @Before
+    public void setup() {
+        importProject();
+    }
+
+    @Test
+    public void testFailImportProjectBuildWithSubPackage() throws CoreException {
+        IProject module1Proj = getWorkspaceRoot().getProject("module1");
+        IProject module4Proj = getWorkspaceRoot().getProject("module4");
+        IProject[] referencedProjects = module1Proj.getReferencedProjects();
+
+        assertFalse("Find module4 in the referenced projects list",
+            Arrays.stream(referencedProjects).anyMatch(proj -> proj.equals(module4Proj)));
```

Comment:
as far as I can comprehend BUILD files for build-with-subpackage workspace module1 is not depending on module4. they both depend on module3.
please correct me if I am wrong